osquery/filesystem/filesystem.cpp

```diff
@@ -139,20 +140,23 @@ Status listFilesInDirectory(const boost::filesystem::path& path,
     boost::filesystem::directory_iterator begin_iter(path);
     boost::filesystem::directory_iterator end_iter;
     for (; begin_iter != end_iter; begin_iter++) {
-      if (!boost::filesystem::is_directory(begin_iter->path())) {
-        results.push_back(begin_iter->path().string());
+      try {
+        if (boost::filesystem::is_regular_file(begin_iter->path())) {
+          results.push_back(begin_iter->path().string());
+        }
+      } catch (const boost::filesystem::filesystem_error& e) {
+        if (ignore_error == 0) {
```

Comment:
You may want to know if there are file in the directory you don't have access to. If we are always ignoring errors, this will just push the error farther down the line to when they try and operate on the file.